discovery_engine_core/web-api/src/handlers.rs

```diff
@@ -122,57 +125,68 @@ pub(crate) async fn handle_personalized_documents(
     }
 
     if all_documents.is_empty() && !errors.is_empty() {
-        return Err(warp::reject::custom(ElasticOpError));
+        return Ok(Box::new(StatusCode::BAD_REQUEST));
```

Comment:
Isn't that an internal server error, because accessing an internal subsystem of ours failed.

Reply:
the [openapi docs](https://spec.openapis.org/oas/v3.1.0#responses-object) is not quite clear to me in that regard:

> The documentation is not necessarily expected to cover all possible HTTP response codes because they may not be known in advance. However, documentation is expected to cover a successful operation response and any known errors.

i'd say that a 500 would be a known error in this case, but our specification doesn't list 500 as a known response, hence i mapped it like this. i guess we can make one of the following decisions here:
- map it as 400 like currently
- extend our spec to include 500 and map it as 500
- implicitely allow 500 as a response in our spec and map it as 500

i have no strong preferences, it just should be handled uniformly in our spec.

Reply:
This is 500 since it does not depends on the client's request. We didn't list 500 because it could always happen and will always be there. We can also add it to the spec in every endpoint but will not add much.

Reply:
changed accordingly.